src/Indicator.vala

```diff
@@ -197,6 +201,9 @@ public class Power.Indicator : Wingpanel.Indicator {
         string? primary_text = null;
         string? secondary_text = null;
         if (display_device != null) {
+            if (display_device.percentage == LOW_BATTERY_PERCENTAGE && !display_device.is_charging) {
+                settings.set_boolean ("show-percentage", true);
```

Comment:
Should revert to not showing the percentage (if that was the original setting) if the device starts charging.  You do not seem to be storing the original setting anywhere?

Reply:
@jeremypw Thanks for your review, I hide the percentage after plug the charger in the last commit